fitbenchmarking/parsing/fitbenchmark_parser.py

```diff
@@ -0,0 +1,259 @@
+"""
+This file implements a parser for the Fitbenchmark data format.
+"""
+
+from __future__ import (absolute_import, division, print_function)
+
+from collections import OrderedDict
+import os
+import mantid.simpleapi as msapi
+import numpy as np
+
+from fitbenchmarking.parsing.base_parser import Parser
+from fitbenchmarking.parsing.fitting_problem import FittingProblem
+from fitbenchmarking.utils.logging_setup import logger
+
+
+class FitbenchmarkParser(Parser):
+    """
+    Parser for the native FitBenchmarking problem definition (FitBenchmark)
+    file.
+    """
+
+    def parse(self):
+        """
+        Parse the Fitbenchmark problem file into a Fitting Problem.
+
+        :return: The fully parsed fitting problem
+        :rtype: fitbenchmarking.parsing.fitting_problem.FittingProblem
+        """
+        fitting_problem = FittingProblem()
+
+        self._entries = self._get_fitbenchmark_data_problem_entries()
+        self._parsed_func = self._parse_function()
+
+        fitting_problem.name = self._entries['name']
+
+        data_points = self._get_data_points()
+
+        fitting_problem.data_x = data_points[:, 0]
+        fitting_problem.data_y = data_points[:, 1]
+        if (data_points.shape[1] > 2):
+            fitting_problem.data_e = data_points[:, 2]
+
+        # String containing the function name(s) and the starting parameter values for each function
+        self._mantid_equation = self._entries['function']
+
+        fitting_problem.functions = self._fitbenchmark_func_definitions()
+
+        # Print number of equations until better way of doing this is looked at.
+        equation_count = len(self._parsed_func)
+        fitting_problem.equation = '{} Functions'.format(equation_count)
+
+        fitting_problem.starting_values = self._get_starting_values()
+
+        # start and end values in x range
+        if 'fit_parameters' in self._entries:
+            start_x, end_x = self._get_x_range()
+            fitting_problem.start_x = start_x
+            fitting_problem.end_x = end_x
+
+        return fitting_problem
+
+    def _get_data_file(self):
```

Comment:
Yes the parsers are virtually identical... and to my knowledge we have creative control to some extent over the formats.
Issue #209 suggests they should be merged. I think now that they have been refactored into what they are now it's pretty clear that they can (and should) be merged.

I guess if I pull out all the matching functionality into a mixin, then when they're merged it will make it extremely easy, although I wont have time to do this today.

Reply:
@Anders-Markvardsen suggested doing this as a separate issue in order to speed up merging of this PR.

Reply:
Agreed, good idea - can you please create the issue for it?

Reply:
Did you get around to creating this issue?